paddle/fluid/imperative/layer.h

```diff
@@ -188,7 +188,29 @@ class VarBase {
     }
   }
 
-  framework::proto::VarType::Type DataType() const { return data_type_; }
+  framework::proto::VarType::Type GetTensorDataType() const {
```

Comment:
How about move this implementation into `VarBase::DataType` instead of adding a new function `VarBase::GetTensorDataType`?

Reply:
Done, thanks.